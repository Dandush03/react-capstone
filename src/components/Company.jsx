import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Header from './Header';
import Details from '../containers/Details';

export default class CompanyDetails extends Component {
  componentDidMount() {
    const { props: { symbol } } = this;
    const title = document.title.split('|');
    document.title = `${title[0]} | ${symbol}`;

    const menu = document.getElementsByTagName('menu')[0];
    const li = menu.getElementsByTagName('li')[2];
    li.classList.add('selected');
  }

  render() {
    const { props: { symbol } } = this;
    return (
      <div className="wrapper">
        <Header />
        <Details symbol={symbol} />
      </div>
    );
  }
}

CompanyDetails.propTypes = {
  symbol: PropTypes.string.isRequired,
};
