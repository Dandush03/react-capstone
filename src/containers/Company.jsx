import React, { Component } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import PropTypes from 'prop-types';


import { handleClosePopUp } from '../actions/popUp';

class Company extends Component {
  constructor(props) {
    super(props);
    this.togglePopup = this.togglePopup.bind(this);
    this.handleRedirection = this.handleRedirection.bind(this);
  }


  togglePopup() {
    const { props: { handleClosePopUp } } = this;
    handleClosePopUp();
  }

  handleRedirection() {
    const { props: { popUp: { data } } } = this;
    const newUrl = `/company/${data.symbol}`;
    window.location.href = newUrl;
  }

  render() {
    const { props: { popUp: { data } }, togglePopup, handleRedirection } = this;
    return (
      <div className="company-popup">
        <div>
          <div className="header-popup">
            <button type="button" onClick={() => togglePopup()}>X</button>
          </div>
          <div className="data-popup">
            <h1>{data.name}</h1>
            <div className="info">
              <div>
                <p>currency</p>
                <h4>{data.currency}</h4>
              </div>
              <div>
                <p>exchange Name</p>
                <h4>{data.exchangeShortName}</h4>
              </div>
              <div>
                <p>stock Exchange</p>
                <h4>{data.stockExchange}</h4>
              </div>
              <div>
                <p>Symbol</p>
                <h4>{data.symbol}</h4>
              </div>
            </div>
          </div>
          <div className="btns">
            <button type="button" onClick={() => togglePopup()}>OK!</button>
            <button type="button" onClick={() => handleRedirection()}>MORE INFO</button>
          </div>
        </div>
      </div>
    );
  }
}

Company.propTypes = {
  handleClosePopUp: PropTypes.func.isRequired,
  popUp: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.bool, PropTypes.object])).isRequired,
};

const structeredSelector = createStructuredSelector({
  popUp: state => state.popUp,
});

const mapDispatchToProps = { handleClosePopUp };

export default connect(structeredSelector, mapDispatchToProps)(Company);
