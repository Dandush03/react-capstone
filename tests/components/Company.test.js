import React from 'react';
import renderer from 'react-test-renderer';

import storeConfig from '../../src/store';
import Company from '../../src/components/Company';


const store = storeConfig();

describe('components', () => {
  describe('Company', () => {
    const component = renderer.create(<Company store={store} />);
    const tree = component.toJSON();
    it('Should match to Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
