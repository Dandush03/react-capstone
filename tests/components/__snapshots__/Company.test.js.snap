// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Company Should match to Snapshot 1`] = `
<div
  className="company-popup"
>
  <div>
    <div
      className="header-popup"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
    <div
      className="data-popup"
    >
      <h1 />
      <div
        className="info"
      >
        <div>
          <p>
            currency
          </p>
          <h4 />
        </div>
        <div>
          <p>
            exchange Name
          </p>
          <h4 />
        </div>
        <div>
          <p>
            stock Exchange
          </p>
          <h4 />
        </div>
        <div>
          <p>
            Symbol
          </p>
          <h4 />
        </div>
      </div>
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      OK!
    </button>
  </div>
</div>
`;
